src/test/crimson/perf_crimson_msgr.cc

```diff
@@ -176,42 +183,93 @@ static seastar::future<> run(
         : public ceph::net::Dispatcher,
           public seastar::peering_sharded_service<Client> {
 
-      struct PingSession : public seastar::enable_shared_from_this<PingSession> {
+      struct ConnSession {
+        mono_time connecting_time = mono_clock::zero();
+        mono_time connected_time = mono_clock::zero();
         unsigned received_count = 0u;
-        mono_time connecting_time;
-        mono_time connected_time;
-        mono_time start_time;
-        mono_time finish_time;
-        seastar::promise<> done;
+
+        mono_time start_time = mono_clock::zero();
+        unsigned start_count = 0u;
+
+        unsigned sampled_count = 0u;
+        double total_lat_s = 0.0;
+
+        // for reporting only
+        mono_time finish_time = mono_clock::zero();
+
+        void start() {
+          start_time = mono_clock::now();
+          start_count = received_count;
+          sampled_count = 0u;
+          total_lat_s = 0.0;
+          finish_time = mono_clock::zero();
+        }
       };
-      using PingSessionRef = seastar::shared_ptr<PingSession>;
+      ConnSession conn_session;
+
+      struct PeriodSession {
```

Comment:
similarly, it's not a session. it's statistics of a given sampling period.

Reply:
Changed to `PeriodStats`